src/main/java/com/deviceinsight/kafka/health/KafkaConsumingHealthIndicator.java

```diff
@@ -74,21 +81,38 @@ public KafkaConsumingHealthIndicator(KafkaHealthProperties kafkaHealthProperties
 		this.consumer = new KafkaConsumer<>(kafkaConsumerPropertiesCopy, deserializer, deserializer);
 		this.producer = new KafkaProducer<>(kafkaProducerProperties, serializer, serializer);
 
-		this.executor = new ThreadPoolExecutor(0, 1, 0L, MILLISECONDS, new SynchronousQueue<>(),
-				new ThreadPoolExecutor.AbortPolicy());
+		this.executor = Executors.newFixedThreadPool(2);
```

Comment:
Why is the second thread needed? What does it do?

Reply:
The second thread is for the continuous polling (to prevent the consumer group timeout).